app-server/app/src/main/scala/user/User.scala

```diff
@@ -20,14 +47,37 @@ object UserService {
 
     database.db.run {
       Users.filter(_.id === id).result.headOption
+    } map {
+      userOption => userOption match {
+        case Some(user) => new UsersRowApi(user)
+        case _ => {
+          // catch this to return 404
+          throw new Exception("404 Resource not found")
```

Comment:
I don't think this is how to do that. Instead returning an `Option[UsersRow]` then in the complete match in the route, if it is `None` return ` complete(404, User not found)` or something similar.